pamelalozano/src/main/webapp/script.js

```diff
@@ -12,24 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
-
+/** Toggles the project's section */
 document.getElementById("projects").addEventListener("click", function (event) {
     if(event.target.classList[0]=="navbar"){
     event.target.closest('.project').classList.toggle("open");
     location.href = "#c-"+event.target.closest('.project').id;
     }
 });
+
+/** Loads existing Comments */
+async function loadComments() {
+  let response = await fetch('/data');
+  let comments = await response.json();
+  if(comments.length==0) {
+      document.getElementById("comments-section").innerHTML="No comments";
+  } 
+  else {
+    let commentSection = document.getElementById("comments-section");
+    
+    //The newest comments at the top
+    for(let i=comments.length-1; 0<=i; i--){
+        let newCard = createCard(comments[i]);
+        commentSection.append(newCard);
+    };
+
+  };
+}
+
+/** Creates a card element containing the comment. */
+function createCard(comment) {
+  let newCard = document.createElement('div');
+  newCard.classList.add('w3-card-4');
+  newCard.append(createCardHeader(comment.subject));
+  newCard.append(createCardComment(comment.msg));
+  newCard.append(createCardFooter(comment.date));
+  return newCard;
+}
+
+/** Creates the comment's header. */
```

Comment:
These three functions are all very similar. Can you create a different helper function that takes a few strings and does the same thing? Might be more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Reply:
If you're planning on header, comment, and footer diverging later, a helper wouldn't be much help.

Reply:
What do you mean by diverging?

Reply:
Right now since the HTML for header, comment and footer looks very similar, it might make sense to refactor the functions into one and pass in parameters like

```js
function createThing(outerType, innerType, text) {
  let container = document.createElement(outerType);
  container.classList.add('w3-container');
  let child = document.createElement(innerType);
  child.innerHTML = text;
  container.append(child);
  return container;
}
```
Use it like
```js
const header = createThing('header', 'h3', headerText);
const comment = createThing('div', 'p', commentText);
```

If they are going to be very different later (diverge from each other), a helper would need more if/else and parameters to handle everything. It might get more complicated than just living with a little repetition. Imagine now you want to add image support to comments. The helper would have to change like this (warning - this is a very contrived example).
```js
function createThing(outerType, innerType, text, imgSrc, thingType) {
  let container = document.createElement(outerType);
  container.classList.add('w3-container');
  container.classList.add(containerClass);
  let child = document.createElement(innerType);
  child.innerHTML = text;
  if (thingType === 'comment' && imgSrc !== null) {
    const img = document.createElement('img');
    img.src = imgSrc;
    child.append(img);
  }
  container.append(child);
  return container;
}
```

I hope it's obvious that the helper function could grow into a monster over time.

Reply:
Ooh Okay I understand, well I'm not sure that i'll keep using that format for the comments I wanted to modify the style but I still don't know how but I think I'll definitely use images an more stuff, I think I'll leave it as it is for now, in case I decide to not change the style I'll make a helper function. :) Thanks